### DISCUSSION TITLE
Will xgterm be substituted by another GUI window?

### DISCUSSION THREAD

**Discussion by Firestar-Reimu** [iraf-community/iraf#394] [General]
As far as I know, iraf commands can run on any terminal emulator, like xterm, konsole, Alacritty, Kitty.

But some GUI window like images and spectra still need xgterm, and the GUI windows must have interaction with keyboard and mouse.

As many new GUI windows have developed and Wayland is substituting X11, also HiDPI screen has been widespread, will xgterm be substituted by another GUI window like the image window of matplotlib?

Thanks!

---

Maybe one day we can see IRAF as an IDE, like Spyder or Pycharm. And we do not need a terminal emulator to run it.

**Comment by olebole** [comment]
IRAF is clearly deprecated software and there is not much enthusiasm except to keep it running until replacements are found for all tasks that are still relevant.

However, it is also Open Source and community maintained, so its fate depends on those who put their efforts into it. If you think there should be a modern replacement for xgterm (aside of PyRAF): go ahead. We'd be happy to maintain it under the iraf-community hood. That is a do-o-cracy; there is no users' committee that has a wishlist and a developer team that implements this, but whoever dedicates time into it will decide the way forward. If nobody puts their efforts, it will not be done.

One idea would be to use [Jupyter Notebooks](https://jupyter.org/) as a base, because it already provides a graphics interface for both vector based and pixel based, which also may be used interactively. Having the PyRAF graphics window integrable into Jupyter would be a first step; separating PyRAF graphics from main PyRAF (and be able to use it with plain CL/ECL) another one.

**Comment by tepickering** [comment]
i am working on https://github.com/tepickering/specterm1d to help fill this need, at least for spectroscopy. it either pops up a graphics window (using matplotlib) or does the plot inline in terminal clients that support it like kitty, wezterm, or ghostty. even xterm supports inline graphics if you specify `xterm -ti vt340`.
